features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5xPalSecurityManager.cpp

```diff
@@ -568,25 +630,26 @@ ble_error_t nRF5xSecurityManager::send_keypress_notification(
 ble_error_t nRF5xSecurityManager::generate_secure_connections_oob(
     connection_handle_t connection
 ) {
-    // FIXME: made with external library; requires SDK v13
-    return BLE_ERROR_NOT_IMPLEMENTED;
-}
+    ble_gap_lesc_p256_pk_t own_secret;
+    ble_gap_lesc_oob_data_t oob_data;
 
+    memcpy(own_secret.pk, secret.buffer(), secret.size());
 
-ble_error_t nRF5xSecurityManager::secure_connections_oob_received(
-    const address_t &address,
-    const oob_lesc_value_t &random,
-    const oob_confirm_t &confirm
-) {
-    // FIXME: store locally
-    return BLE_ERROR_NOT_IMPLEMENTED;
-}
+    uint32_t err = sd_ble_gap_lesc_oob_data_get(
```

Comment:
Didn't you say this was really slow? doesn't this block?

Reply:
That's not the slower part; computing the shared secret is. 
This does block application code but we have no choice outside spawning a worker thread which is not acceptable at the current minute.